### PR TITLE
Add Unit Tests for coverage for FreeRTOS_DNS_Parser

### PR DESCRIPTION
**Pull request by kar-rahul-aws**
UT for DNS_Parser

Description
-----------
UT added in DNS_Parser for 100% coverage

Test Steps
-----------
```
cmake -S test/unit-test -B test/unit-test/build/ -DCMAKE_BUILD_TYPE=Debug
make -C test/unit-test/build/ all
```
Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [x] I have modified and/or added unit-tests to cover the code changes in this Pull Request.

Related Issue
-----------

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: source/FreeRTOS_DNS_Parser.c [hunk 1]
```diff
@@ -1243,10 +1230,14 @@
...
-                    vReleaseNetworkBufferAndDescriptor( pxNewBuffer );
-                }
+/*pxNewBuffer and pxNetworkBuffer are now the same pointers.pxNewBuffer is always NULL after re-allocation*/
+
+/*
+ *              if( pxNewBuffer != NULL )
+ *              {
+ *                vReleaseNetworkBufferAndDescriptor( pxNewBuffer );
+ *              }
+ */
```
Comment (ActoryOu) [inline source/FreeRTOS_DNS_Parser.c:1240]: Can you elaborate more about why pxNewBuffer is always NULL?
It's used in different APIs as network buffer and it's weird to be NULL.

File: test/unit-test/FreeRTOS_DNS_Parser/FreeRTOS_DNS_Parser_stubs.c [hunk 2]
```diff
@@ -54,6 +54,13 @@ void vPortExitCritical( void )
 
 uint16_t usPacketIdentifier;
 
+
```
Comment (ActoryOu) [inline test/unit-test/FreeRTOS_DNS_Parser/FreeRTOS_DNS_Parser_stubs.c:57]: Please follow the file format in stub.c as well.

File: test/unit-test/FreeRTOS_DNS_Parser/FreeRTOS_DNS_Parser_utest.c [hunk 3]
```diff
@@ -80,6 +80,9 @@ static void dns_callback( const char * pcName,
 
 extern BaseType_t xBufferAllocFixedSize;
 
+extern pucAddrBuffer[ 2 ];
```
Comment (ActoryOu) [inline test/unit-test/FreeRTOS_DNS_Parser/FreeRTOS_DNS_Parser_utest.c:83]: Please follow the file format in _utest.c

**Review by ActoryOu — CHANGES_REQUESTED** [review]

File: test/unit-test/FreeRTOS_DNS_Parser/FreeRTOS_DNS_Parser_utest.c (hunk 3 above)
Comment (kar-rahul-aws) [reply test/unit-test/FreeRTOS_DNS_Parser/FreeRTOS_DNS_Parser_utest.c:83]: Updated.

File: test/unit-test/FreeRTOS_DNS_Parser/FreeRTOS_DNS_Parser_stubs.c (hunk 2 above)
Comment (kar-rahul-aws) [reply test/unit-test/FreeRTOS_DNS_Parser/FreeRTOS_DNS_Parser_stubs.c]: Updated.

File: source/FreeRTOS_DNS_Parser.c (hunk 1 above)
Comment (kar-rahul-aws) [reply source/FreeRTOS_DNS_Parser.c]: Updated the comment, pxNewBuffer and pxNetworkBuffer are the same buffer and not duplicated, hence it is enough to Release the buffer once, no need for the If condition check

**Review by ActoryOu — APPROVED** [review]

**Review by shubnil — APPROVED** [review]

**Pull request merged**